src/v2/models/AppState.js

```diff
@@ -105,9 +105,33 @@ export default Model.extend({
   },
 
   shouldReRenderView (transformedResponse) {
+    _.mixin({
+      nestedOmit: function (obj, iteratee, context) {
+        var result = _.omit(obj, iteratee, context);
+
+        _.each(result, function (val, key) {
+          if (typeof(val) === 'object') {
+            result[key] = _.nestedOmit(val, iteratee, context);
+          }
+        });
+
+        return result;
+      }
+    });
+
     const previousRawState = this.has('idx') ? this.get('idx').rawIdxState : null;
-    const identicalResponse = _.isEqual(_.omit(transformedResponse.idx.rawIdxState, 'expiresAt'),
+    const identicalResponse = _.isEqual(_.nestedOmit(transformedResponse.idx.rawIdxState, ['expiresAt', 'refresh']),
+      _.nestedOmit(previousRawState, ['expiresAt', 'refresh']));
+    const noPollRefreshNeeded = _.isEqual(_.omit(transformedResponse.idx.rawIdxState, 'expiresAt'),
       _.omit(previousRawState, 'expiresAt') );
+
+    if (identicalResponse && !noPollRefreshNeeded) {
+      const currentFormName = this.get('currentFormName');
+      const currentViewState = transformedResponse.remediations.filter(r => r.name === currentFormName)[0];
+
+      this.trigger('refreshUpdated', currentViewState);
```

Comment:
I don't remember the reason to trigger an event from here ? can't be updating `reRender` to `false` is suffice ?

Reply:
Since the SIW will not rerender with these new nested omit changes, the SIW won't be updating the polling interval value anymore in DeviceChallengePollView, meaning it'll keep polling the server at the initial fast `refresh` interval. So, this event is used to trigger that update.

Reply:
okay that make sense... Let's try another approach instead.. 
1. add a local boolean attribute in line 23 as `dynamicRefresh`
2. set it's value to true/false accordingly in `shouldReRenderView` 
3. Listen to change `change` event on this property in your DeviceChallengePollView and update the interval